### PR TITLE
Add README

### LINKED ISSUE #8: minimize dependencies
Followup issue for https://github.com/scalameta/metals/issues/285

Logback is replaced with scribe, but an effort to replace circe and cats in #4 was abandoned. For distribution in clients, a smaller library footprint is very desirable.

Olaf suggested alot of the bulk can be removed by pruning unused cats implicits.

### PR DESCRIPTION
**Pull request by olafurpg**
- Remove Enumeratum dependency in jsonrpc module. It was only
  used for ErrorCode and then it turned out that the JSON
  decoder for `ErrorCode` was buggy since it would fail on
  unknown error codes.
- Document dependencies
- Installation instructions
- Gettings started instructions
- Document alternatives

Fixes #8

### CODE REVIEW

**Comment by olafurpg** [comment]
PS @jvican and @jastice, we should update the `bsp` module to depend on `jsonrpc` instead of `lsp4s`. That should shave off around 1.5mb, in combination with removing shapeless in #9, pprint in #10 and enumeratum in this PR the next bsp release could be ~5mb smaller.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,213 @@
...
+[lsp]: https://microsoft.github.io/language-server-protocol/
+[monix]: https://monix.io/
+[circe]: https://circe.github.io/circe/
+[scribe]: https://github.com/outr/scribe
+
+## Getting Started
+
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12/badge.svg)](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12)
+
+To implement a language server, add the following dependency you need.
```
Comment (gabro) [inline README.md]: This sounds like Yoda 😅

**Review by gabro — APPROVED** [review]
👌 two minor comments.
Thanks for writing this, it's extremely useful!

File: README.md (hunk 1 above)
Comment (olafurpg) [reply README.md]: Haha, good catch, was a bad rephrase from an old text

**Comment by olafurpg** [comment]
Weird, the tests started failing consistently on 2.12 only and it seems unrelated to the PR 🤔 will need to investigate...

**Comment by jvican** [comment]
I’ll try to review tonight, if I’m blocking something feel free to merge!

**Comment by olafurpg** [comment]
There's no rush @jvican

**Comment by olafurpg** [comment]
My conclusion is that the ping pong suite was flaky and it just happened to pass the CI in #10. The tests failed in two different ways

- Timeout, all tests had passed but the sbt shell never exited

```
[info] Passed: Total 21, Failed 0, Errors 0, Passed 21
[info] Done compiling.
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

- Assertion error: the second pong response never delivered

```
[info] - ping pong *** FAILED ***
[info]   received List(Pong from client) != expected List(Pong from client, Pong from server) (PingPongSuite.scala:68)
```

I've refactored in the ping pong suite to

- ensure that we close all input/output streams from the test connection. This should fix the timeout error.
- ensure that the assertions run after all pongs have been received by using `Promise.complete`, previously we relied on  the the scheduling from the execution context. This should fix the assertion error.

While refactoring, I found it awkward to read `connection.client.client`/`connection.server.client`. Given that JSON-RPC is bi-directional I renamed `connection.client` to `connection.alice` and `connection.server` to `connection.bob`.

I've restarted the CI build three times now and the tests have always passed.

**Comment by olafurpg** [comment]
Merging to fix flaky tests on master. Feel free to follow up later once you have time to review @jvican

**Pull request merged**